src/modules/rc_update/rc_update.cpp

```diff
@@ -687,6 +692,7 @@ void RCUpdate::UpdateManualControlInput(const hrt_abstime &timestamp_sample)
 	manual_control_input.aux4  = get_rc_value(rc_channels_s::FUNCTION_AUX_4,   -1.f, 1.f);
 	manual_control_input.aux5  = get_rc_value(rc_channels_s::FUNCTION_AUX_5,   -1.f, 1.f);
 	manual_control_input.aux6  = get_rc_value(rc_channels_s::FUNCTION_AUX_6,   -1.f, 1.f);
+	manual_control_input.valid = _rc_calibrated && (_param_rc_chan_cnt.get() == _channel_count_previous);
```

Comment:
I'm not sure about this `_param_rc_chan_cnt.get() == _channel_count_previous`. If either the param changes or the count for some reason in flight, you lose RC.

Reply:
Thanks for reviewing! Yes, I thought about that. But extending that thought if the calibration changes in air you might have malicious input e.g. pulling the vehicle in an undesired direction. In any case, I'll drop this check for now and we can add it at least as a warning in air with the stick input workflow changes.